src/charm.py

```diff
@@ -253,8 +253,14 @@ def _on_storage_attached(self, event: StorageAttachedEvent) -> None:
         if self.workload.active():
             self._set_status(Status.ADDED_STORAGE)
             self.workload.exec(f"chown -R {USER}:{GROUP} {self.workload.paths.data_path}")
-            self.workload.exec(f"chmod -R 770 {self.workload.paths.data_path}")
+            self.workload.exec(f"chmod -R 750 {self.workload.paths.data_path}")
             self._on_config_changed(event)
+        else:
+            self.workload.exec(f"chmod -R 750 {self.workload.paths.data_path}")
+            self.workload.exec(f"chown -R {USER}:{GROUP} {self.workload.paths.data_path}")
+            self.workload.exec(
+                f"""find {self.workload.paths.data_path} -type f -name "meta.properties" -delete || true"""
```

Comment:
**question** what is this for?

Reply:
This removes `meta.properties` files in `$DATA` subfolders (plural because multiple storages can be attached to the VM charm). This file contains the `broker.id=<old_broker_id>` line that causes the mismatch